openpype/modules/extra_addons/process_monitor/addon.py

```diff
@@ -0,0 +1,704 @@
+"""
+
+psutil 'process_iter' parameters:
+'cmdline', 'connections', 'cpu_affinity', 'cpu_num', 'cpu_percent',
+'cpu_times', 'create_time', 'cwd', 'environ', 'exe', 'gids', 'io_counters',
+'ionice', 'memory_full_info', 'memory_info', 'memory_maps', 'memory_percent',
+'name', 'nice', 'num_ctx_switches', 'num_fds', 'num_handles', 'num_threads',
+'open_files', 'pid', 'ppid', 'status', 'terminal', 'threads', 'uids', 'username'
+
+"""
+
+import os
+import click
+
+import threading
+import ctypes
+import platform
+
+from openpype.lib import Logger
+
+#TODO: Test with Python2
+import psutil
+
+
+from openpype.modules import (
+    JsonFilesSettingsDef,
+    OpenPypeAddOn,
+    ModulesManager,
+    IPluginPaths,
+    ITrayAction
+)
+
+from openpype.api import get_anatomy_settings, get_system_settings
+
+
+#TODO: Add settings (?)
+# Settings definition of this addon using `JsonFilesSettingsDef`
+# - JsonFilesSettingsDef is prepared settings definition using json files
+#   to define settings and store default values
+class AddonSettingsDef(JsonFilesSettingsDef):
+    # This will add prefixes to every schema and template from `schemas`
+    #   subfolder.
+    # - it is not required to fill the prefix but it is highly
+    #   recommended as schemas and templates may have name clashes across
+    #   multiple addons
+    # - it is also recommended that prefix has addon name in it
+    schema_prefix = "process_monitor"
+
+    def get_settings_root_path(self):
+        """Implemented abstract class of JsonFilesSettingsDef.
+
+        Return directory path where json files defying addon settings are
+        located.
+        """
+        return os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "settings"
+        )
+
+
+class ProcessMonitor(OpenPypeAddOn, IPluginPaths, ITrayAction):
```

Comment:
I'm not 100% sure it's the right name. Suggesting `ProcessTracker` which would suit it better. It's still based on matching executables from settings and not really monitoring launched applications on deterministic logic.